provider/grpc/provider.go

```diff
@@ -55,7 +55,9 @@ func (g *grpcProvider) ProviderInit(ctx context.Context) error {
 }
 
 func (g *grpcProvider) Capabilities() []provider.Capability {
-	r, err := g.Client.Capabilities(context.TODO(), &emptypb.Empty{})
+	// Capabilities requests happen at the start of the analyzer-lsp, so it makes
+	// sens to wait until the connection is ready.
+	r, err := g.Client.Capabilities(context.TODO(), &emptypb.Empty{}, grpc.WaitForReady(true))
```

Comment:
I think that if you use wait for ready, that you need to set a timeout 

```. Otherwise, the RPC client will block the call until a connection is available (or the call is canceled or times out) and will retry the call if it fails due to a transient error.```

I also wonder if making the log a deeper verbosity and keeping the call as is, would allow us to debug the case when the server is not ready easier.

Just some thoughts and questions WDYT?

Reply:
Didn't see the blocking thing with `WaitForReady`, I'll add a timeout! Technically, this could be solved by putting a `time.sleep` just before calling this, but I wanted a more robust solution. I also agree with making the verbosity deeper

I couldn't find a way to wait/check for the grpc server to be ready after instantiation. I think that'd probably be the best way to do it if we could find it

Reply:
The timeout can be added to the context.

I would probably prefer to use our retry logic (that calls the caps and the reason you get this message every time) and reduce the logging verbosity. This will allow us to turn up the verbosity and see if the server never becomes active. WDYT?